tycho-extras/tycho-p2-extras-plugin/src/main/java/org/eclipse/tycho/plugins/p2/extras/CompareWithBaselineMojo.java

```diff
@@ -106,6 +106,17 @@ public static enum ReportBehavior {
     @Parameter(property = "onIllegalVersion", defaultValue = "fail", alias = "tycho.p2.baseline.onIllegalVersion")
     private ReportBehavior onIllegalVersion;
 
+    /**
+     * The version range of the reference artifact in the baseline.
+     * <p>
+     * This can be used to select the desired baseline artifact if multiple versions of an artifact
+     * are available and to compare. By default a local artifact is compared to the latest version
+     * in the baseline.
+     * </p>
+     */
+    @Parameter(defaultValue = "0.0.0", alias = "tycho.p2.baseline.versionRange")
+    private String baselineVersionRange = "0.0.0";
```

Comment:
This seems hardly usable, as one needs to explicitly configure the version per project (also `0.0.0` is not really a version range).

In PDE features we have the [concept of a "match rule"](https://help.eclipse.org/latest/topic/org.eclipse.pde.doc.user/guide/tools/editors/feature_editor/feature_editor.htm?cp=4_3_2_1) that might be more sufficient here as we then can compute a version range based on the current project version.

Reply:
Yes that's what I meant in my initial comment with:
> Instead of specifying the version range explicitly we could also define an enum similar to ReportBehavior to specify 'named version rules' for the baseline artifact, like `same-major` and `latest`.

So I think the main question is what rules to support?
We could start simple and just have something like `same-major` and `latest`. Using the exact set of rules from PDE doesn't look suitable to me, because `Perfect` and `Greater or Equal` doesn't make sense in this context, because this mojo also checks for backward steps.
So besides `same-major` and `latest` actually only `same-minor` might be of interest.
But the names of the rules could be more speaking.

For the implementation we maybe could re-use PDE's new:
https://github.com/eclipse-pde/eclipse.pde/blob/3f973e87fd7ede513388e8ed870d1b65de438947/ui/org.eclipse.pde.core/src/org/eclipse/pde/core/plugin/VersionMatchRule.java

Reply:
> So I think the main question is what rules to support?

We can just support what is useful today and add more if required, I think thats how its already done on some other places in Tycho

> We could start simple and just have something like same-major and latest.

I would prefer to just use the same naming as for features as this is a P2 concept and we should not mix to much naming.

>  Using the exact set of rules from PDE doesn't look suitable to me, because Perfect and Greater or Equal doesn't make sense in this context, 

Greater or equal is actually what is the default today, the only one not so useful might be "Perfect" that actually would never match anything.

> For the implementation we maybe could re-use PDE's new:

Tycho has already `org.eclipse.tycho.model.Feature.getVersionRange(ImportRef)` we can simply generalize that method, so I don't think relying on PDE-UI in Tycho just to reuse an enum is very useful, if we ever do
- https://github.com/eclipse-pde/eclipse.pde/issues/34

it would of course be good to reuse as much as possible.